### PR TITLE
Couple of updates

### PR DESCRIPTION
**Pull request by mikesir87**
Couple of updates

- Update the `words` service to use Java 17 (which is LTS). Specifically picked Amazon Corretto due to its maintenance and support
- Removed the usage of the custom db image, since the image was only baking in the schema files. We can simply mount those in the Compose and k8s manifests
- Updated the k8s manifests to use Kustomize to create a ConfigMap using the same db schema file

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,19 +1,19 @@
 version: '3.3'
```
Comment (BretFisher) [inline docker-compose.yml:1]: I recommend commenting out this line and putting in a similar statement to what I put in all my teaching compose files for 2020 and beyond:

```text
# `docker-compose` CLI is now replaced with `docker compose`
# no version key:value needed in these files
# old compose file versions (2.x and 3.x) are superseded by the "compose spec"
# more info https://github.com/compose-spec/compose-spec
```

**Review by BretFisher — APPROVED** [review]
I recommend removing compose version before merging.

File: docker-compose.yml (hunk 1 above)
Comment (mikesir87) [reply docker-compose.yml:1]: I tried to make only the minimal updates for now, as we need to make another pass to have the compose file actually be a dev-ready compose file (watch for file changes, recompile, etc.).

Comment (mikesir87) [reply docker-compose.yml:1]: So, figured that next round could include the version update, etc. But, happy to do so now if you'd prefer that.

**Pull request merged**